waku/node/v2/wakunode2.nim

```diff
@@ -182,15 +183,12 @@ proc publish*(node: WakuNode, topic: Topic, message: WakuMessage) =
   # XXX Consider awaiting here
   discard wakuRelay.publish(topic, data)
 
-proc query*(w: WakuNode, query: HistoryQuery): HistoryResponse =
-  ## Queries for historical messages.
+proc query*(w: WakuNode, query: HistoryQuery, handler: QueryHandlerFunc) {.async, gcsafe.} =
```

Comment:
See above.

What happens if it fails? Is there a timeout? May it return an empty response, and what are the semantics if so?

Reply:
Currently no thought has been given towards that. I think it makes sense to add in later iterations after this is merged and we refactor the way the RPC works.